cdap-common/src/test/java/co/cask/cdap/common/service/RetryableScheduledServiceTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.service;
+
+import co.cask.cdap.api.retry.RetriesExhaustedException;
+import co.cask.cdap.common.utils.Tasks;
+import com.google.common.base.Throwables;
+import com.google.common.util.concurrent.Service;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+/**
+ * Unit test class for the {@link AbstractRetryableScheduledService}.
+ */
+public class RetryableScheduledServiceTest {
+
+  @Test
+  public void testNormalIteration() throws InterruptedException {
+    CountDownLatch latch = new CountDownLatch(5);
+
+    Service service = new AbstractRetryableScheduledService(RetryStrategies.noRetry()) {
+      @Override
+      protected long runTask() {
+        latch.countDown();
+        return 1L;
+      }
+    };
+
+    service.start();
+    Assert.assertTrue(latch.await(5, TimeUnit.SECONDS));
+    service.stopAndWait();
+  }
+
+  @Test
+  public void testRetryExhausted() throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
name is swapped with `testNoRetry`?

Reply:
Actually the naming are correct. This method is for testing the retry strategy exhausted retrying (the noRetry strategy is used to make it exhausted immediately). The `testNoRetry` is for testing the case where a service decided not to retry on a given `Exception`.